tests/new_functional/spec/navigation.spec.js

```diff
@@ -0,0 +1,204 @@
+        .click('a=Household Composition')
+        .click(HouseholdCompositionPage.addPerson())
+        .setValue(HouseholdCompositionPage.firstName('_1'), 'Another')
+        .setValue(HouseholdCompositionPage.lastName('_1'), 'User')
+        .click(HouseholdCompositionPage.submit())
+        .isVisible('a=Test User').should.eventually.be.true
+        .isVisible('a=Another User').should.eventually.be.true
+        .click('a=Another User')
+        .getUrl().should.eventually.contain(RepeatingBlock1Page.pageName);
+      });
+  });
+
+  it('Given I add multiple people to Household Composition, completed status should be independent', function() {
+    return helpers.openQuestionnaire('test_navigation.json').then(() => {
+      return browser
+        .click('a=Household Composition')
+        .setValue(HouseholdCompositionPage.firstName(), 'Test')
+        .setValue(HouseholdCompositionPage.lastName(), 'User')
+        .click(HouseholdCompositionPage.submit())
+        .click('a=Household Composition')
+        .click(HouseholdCompositionPage.addPerson())
+        .setValue(HouseholdCompositionPage.firstName('_1'), 'Another')
+        .setValue(HouseholdCompositionPage.lastName('_1'), 'User')
+        .click(HouseholdCompositionPage.submit())
+        .isVisible('a=Test User').should.eventually.be.true
+        .isVisible('a=Another User').should.eventually.be.true
+        .click(RepeatingBlock1Page.yes())
+        .click(RepeatingBlock1Page.submit())
+        .click(RepeatingBlock2Page.employed())
+        .checkCompletedTick('a=Test User').should.eventually.be.false
+        .checkCompletedTick('a=Another User').should.eventually.be.false
+        .click(RepeatingBlock2Page.submit())
+        .checkCompletedTick('a=Test User').should.eventually.be.true
+        .checkCompletedTick('a=Another User').should.eventually.be.false;
+      });
+  });
+
+  it('Given I add then remove people to Household Composition, additional People links should disappear', function() {
+    return helpers.openQuestionnaire('test_navigation.json').then(() => {
+      return browser
+        .click('a=Household Composition')
+        .setValue(HouseholdCompositionPage.firstName(), 'Test')
+        .setValue(HouseholdCompositionPage.lastName(), 'User')
+        .click(HouseholdCompositionPage.addPerson())
+        .setValue(HouseholdCompositionPage.firstName('_1'), 'Another')
+        .setValue(HouseholdCompositionPage.lastName('_1'), 'User')
+        .click(HouseholdCompositionPage.submit())
+        .isVisible('a=Test User').should.eventually.be.true
+        .isVisible('a=Another User').should.eventually.be.true
+        .click('a=Household Composition')
+        .click(HouseholdCompositionPage.removePerson())
+        // Wait until page refresh
+        .waitUntil(function () {
+          return browser.elements('.question__answers > div').then(function (e) {
+            return e.value.length === 1;
+          });
+        }, 2000, 'Person not removed in time')
```

Comment:
Not sure why we need to wait for the person to be removed before submitting. Is this an issue we need to look into as it could affect users completing the survey?